tests/model/writer.js

```diff
@@ -1952,128 +1952,159 @@ describe( 'Writer', () => {
+			expect( () => {
+				addMarker( 'name', { usingOperation: false } );
+			} ).to.throw( CKEditorError, /^writer-addMarker-no-range/ );
+		} );
 
-			model.on( 'applyOperation', spy );
+		it( 'should throw when trying to use detached writer', () => {
+			const marker = addMarker( 'name', { range, usingOperation: false } );
+			const writer = new Writer( model, batch );
 
-			setMarker( marker, { usingOperation: true } );
+			expect( () => {
+				writer.addMarker( marker, null, { usingOperation: true } );
+			} ).to.throw( CKEditorError, /^writer-incorrect-use/ );
+		} );
+	} );
 
-			const op = batch.deltas[ 0 ].operations[ 0 ];
+	describe( 'updateMarker()', () => {
+		let root, range;
 
-			expect( spy.calledOnce ).to.be.true;
-			expect( spy.firstCall.args[ 1 ][ 0 ].type ).to.equal( 'marker' );
-			expect( op.oldRange ).to.be.null;
-			expect( op.newRange.isEqual( range ) ).to.be.true;
+		beforeEach( () => {
+			root = doc.createRoot();
+			root._appendChildren( new Text( 'foo' ) );
+			range = Range.createIn( root );
 		} );
 
-		it( 'should create a unique id if the first param is type of range', () => {
-			const marker = setMarker( range );
+		it( 'should accept marker instance', () => {
+			const marker = addMarker( 'name', { range, usingOperation: true } );
+			const range2 = Range.createFromParentsAndOffsets( root, 0, root, 0 );
+
+			updateMarker( marker, { range: range2, usingOperation: true } );
+
+			expect( batch.deltas.length ).to.equal( 2 );
 
-			expect( marker.name ).to.be.a( 'string' );
+			const op = batch.deltas[ 1 ].operations[ 0 ];
+
+			expect( marker.getRange().isEqual( range2 ) ).to.be.true;
+			expect( op.oldRange.isEqual( range ) ).to.be.true;
+			expect( op.newRange.isEqual( range2 ) ).to.be.true;
 		} );
 
-		it( 'should create a unique id if the first param is type of range when usingOperation is set to true', () => {
+		it( 'should not use operations when updating marker which does not use operations', () => {
 			const spy = sinon.spy();
 			model.on( 'applyOperation', spy );
 
-			const marker = setMarker( range, { usingOperation: true } );
+			const marker = addMarker( 'name', { range, usingOperation: false } );
+			const range2 = Range.createFromParentsAndOffsets( root, 0, root, 0 );
+
+			updateMarker( marker, { range: range2, usingOperation: false } );
 
-			expect( marker.name ).to.be.a( 'string' );
-			expect( spy.calledOnce ).to.be.true;
+			sinon.assert.notCalled( spy );
 		} );
 
-		it( 'should use operations when having set usingOperation to true', () => {
+		it( 'should accept empty range parameter if marker instance is passed and usingOperation is set to true', () => {
+			const marker = addMarker( 'name', { range, usingOperation: true } );
 			const spy = sinon.spy();
 
 			model.on( 'applyOperation', spy );
 
-			setMarker( 'name', range, { usingOperation: true } );
+			updateMarker( marker, { usingOperation: true } );
```

Comment:
You don't change range nor `usingOperation`. I do not get what this test is about.